python/ray/dashboard/client/src/pages/dashboard/logical-view/Actor.tsx

```diff
@@ -181,14 +215,106 @@ class Actor extends React.Component<Props & WithStyles<typeof styles>, State> {
-                Object.entries(actor.requiredResources).length > 0 &&
-                Object.entries(actor.requiredResources)
-                  .sort((a, b) => a[0].localeCompare(b[0]))
-                  .map(([key, value]) => `${value.toLocaleString()} ${key}`)
-                  .join(", "),
+                Object.entries(actor.requiredResources).length > 0
+                  ? Object.entries(actor.requiredResources)
+                      .sort((a, b) => a[0].localeCompare(b[0]))
+                      .map(([key, value]) => `${value.toLocaleString()} ${key}`)
+                      .join(", ")
+                  : "",
             },
           ];
 
+    // Apply transformation to add styling for information field
+    const transforms: Record<
+      string,
+      (info: ActorInformation) => ActorInformationRendered
+    > = {
+      Pending: (info: ActorInformation) => {
+        if (actor.state !== -1 && actor.taskQueueLength >= 50) {
+          return {
+            label: info.label,
+            rendered: (
+              <span
+                className={classes.warningTooManyPendingTask}
+                key={info.label}
+              >
+                {info.label}: {info.value}
+              </span>
+            ),
+          };
+        } else {
+          return { label: info.label, rendered: info.label };
+        }
+      },
+      ActorTitle: (info: ActorInformation) => ({
+        rendered: (
+          <span className={classes.actorTitle} key={info.label}>
+            {info.value}
+          </span>
+        ),
+        label: info.label,
+      }),
+      NumObjectIdsInScope: (info: ActorInformation) => ({
+        rendered: (
+          <span className={classes.secondaryFieldsHeader} key={info.label}>
+            <br></br>
+            {info.label}: {info.value}
+          </span>
+        ),
+        label: info.label,
+      }),
+      NumLocalObjects: (info: ActorInformation) => ({
+        rendered: (
+          <span className={classes.secondaryFields} key={info.label}>
+            {info.label}: {info.value}
+          </span>
+        ),
+        label: info.label,
+      }),
+      UsedLocalObjectMemory: (info: ActorInformation) => ({
+        rendered: (
+          <span className={classes.secondaryFields} key={info.label}>
+            {info.label}: {info.value}
+          </span>
+        ),
+        label: info.label,
+      }),
+      Identity: (info: ActorInformation) => ({
+        rendered:
+          info.value.length === 0 ? null : (
+            <span className={classes.datum}>
+              {info.label}: {info.value}
+            </span>
+          ),
+        label: info.label,
+      }),
+    };
+
+    // Apply the styling transformation
+    const information = rawInformation.map(val => {
+      const transform = transforms[val.label];
+      if (transform !== undefined) {
+        return transform(val);
+      } else {
+        return transforms["Identity"](val);
+      }
+    });
```

Comment:
To simplify things a bit, since all of the transforms pass the label through untouched, why don't we have the transform just map from label to rendered element (so it will just return a React node rather than an object), and take the label from `val` instead?

Reply:
That works, but i need to move the reordering part up